[sentencepiece] Clarify supported triplets

Thomas1664 opened the pull request:
**Describe the pull request**

- #### What does your PR fix?  
  Fixes #23436

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?  
  Only windows-static, Linux and Macos are supportet.

  Removed sentencepiece from CI baseline:
```
sentencepiece:arm64-windows=fail
sentencepiece:arm-uwp=fail
sentencepiece:x64-uwp=fail
sentencepiece:x64-windows=fail
sentencepiece:x86-windows=fail
```

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?  
  Yes

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?  
 Yes, but still working on this PR

Thomas1664 committed:
Set supports expression

Thomas1664 committed:
version

Thomas1664 committed:
Fix typo

Thomas1664 committed:
version

Thomas1664 commented:
Reopening because CI failures aren't my fault

FrankXie05 commented:
@Thomas1664 Thanks for posting this PR, but I would like to ask you why you need to modify the support in `ci.baseline.txt`? Did it change upstream?

FrankXie05 commented:
/azp run

Thomas1664 commented:
> @Thomas1664 Thanks for posting this PR, but I would like to ask you why you need to modify the support in `ci.baseline.txt`? Did it change upstream?

@FrankXie05 I added `supports` in `vcpkg.json` to make it more obvious that this ports only supports static linking. Because of `supports` I removed it from CI baseline.

FrankXie05 commented:
@Thomas1664 got it, thanks for your contribution. :)
